### PR TITLE
Docs: Improving the install with `keadm` documentation

### PR DESCRIPTION
**Pull request by 1Shubham7**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
docs update

this is one among series of PRs planned to improve and rewrite some parts of documentation.

Works towards solving #533

### CODE REVIEW

**Comment by 1Shubham7** [comment]
@fisherxu @wbc6080, I need to know whether we should be using "Keadm" or "keadm" consistently in the docs.

File: docs/setup/install-with-keadm.md [hunk 1]
```diff
@@ -2,51 +2,54 @@
...
-Keadm is used to install the cloud and edge components of KubeEdge. It is not responsible for installing K8s and runtime.
 
-Please refer [kubernetes-compatibility](https://github.com/kubeedge/kubeedge#kubernetes-compatibility) to get **Kubernetes compatibility** and determine what version of Kubernetes would be installed.
+Keadm is used to install the cloud and edge components of KubeEdge. It does not handle the installation of Kubernetes and its runtime environment.
 
-## Limitation
+Please refer to [Kubernetes compatibility](https://github.com/kubeedge/kubeedge#kubernetes-compatibility) documentation to check **Kubernetes compatibility** and ascertain the Kubernetes version to be installed.
 
-- Need super user rights (or root rights) to run.
+## Limitation
```
Comment (fujitatomoya) [inline docs/setup/install-with-keadm.md:10]: i think this is not limitation but more like condition. how about this?

```suggestion
## Prerequisite
```

File: docs/setup/install-with-keadm.md [hunk 2]
```diff
@@ -2,51 +2,54 @@
...
-- Download from [github release](https://github.com/kubeedge/kubeedge/releases).
+1. Download from [GitHub release](https://github.com/kubeedge/kubeedge/releases).
   
-    Now KubeEdge github officially holds three arch releases: amd64, arm, arm64. Please download the right arch package according to your platform, with your expected version.
+    KubeEdge GitHub officially holds three architecture releases: amd64, arm, and arm64. Please download the correct package according to your platform and desired version.
+
     ```shell
     wget https://github.com/kubeedge/kubeedge/releases/download/v1.12.1/keadm-v1.12.1-linux-amd64.tar.gz
     tar -zxvf keadm-v1.12.1-linux-amd64.tar.gz
     cp keadm-v1.12.1-linux-amd64/keadm/keadm /usr/local/bin/keadm
```
Comment (fujitatomoya) [inline docs/setup/install-with-keadm.md:25]: we are not tracking this example version, but v1.12.1 seems pretty old, can we instead refer to v1.16.2 at this time?

```suggestion
    wget https://github.com/kubeedge/kubeedge/releases/download/v1.16.2/keadm-v1.16.2-linux-amd64.tar.gz
    tar -zxvf keadm-v1.16.2-linux-amd64.tar.gz
    cp keadm-v1.16.2-linux-amd64/keadm/keadm /usr/local/bin/keadm
```

File: docs/setup/install-with-keadm.md [hunk 3]
```diff
@@ -2,51 +2,54 @@
...
+3. Build from Source
   
-    ref: [build from source](./install-with-binary#build-from-source)
-
+-   Refer to [build from source](./install-with-binary#build-from-source) for instructions.
 
 ## Setup Cloud Side (KubeEdge Master Node)
 
-By default ports `10000` and `10002` in your cloudcore needs to be accessible for your edge nodes.
+By default, ports `10000` and `10002` on your CloudCore needs to be accessible for your edge nodes.
```
Comment (fujitatomoya) [inline docs/setup/install-with-keadm.md:40]: so the `CloudCore` is the correct spelling? i was thinking those are `cloudcore` and `edgecore`.

File: docs/setup/install-with-keadm.md [hunk 4]
```diff
@@ -152,18 +163,22 @@ Run `keadm gettoken` in **cloud side** will return the token, which will be used
 ### Join Edge Node
 
 #### keadm join
-`keadm join` will install edgecore. It also provides a flag by which a specific version can be set. It will pull image [kubeedge/installation-package](https://hub.docker.com/r/kubeedge/installation-package) from dockerhub and copy binary `edgecore` from container to hostpath, and then start `edgecore` as a system service.
+
+`keadm join` installs Edgecore. It also provides a flag  to set a specific version. It pulls the image [kubeedge/installation-package](https://hub.docker.com/r/kubeedge/installation-package) from Docker Hub, copies the `edgecore` binary from container to the hostpath, and then starts `edgecore` as a system service.
```
Comment (fujitatomoya) [inline docs/setup/install-with-keadm.md:167]: same here, i thought that is `edgecore`, that is the question here.

**Review by fujitatomoya — COMMENTED** [review]
overall, ltgm. maybe we can ask for maintainers about names of terminologies.

File: docs/setup/install-with-keadm.md (hunk 3 above)
Comment (1Shubham7) [reply docs/setup/install-with-keadm.md:40]: at some places it was cloudcore and at some it was Cloudcore. should it be "cloudcore"?

File: docs/setup/install-with-keadm.md (hunk 2 above)
Comment (1Shubham7) [reply docs/setup/install-with-keadm.md:25]: I agree!

File: docs/setup/install-with-keadm.md [hunk 5]
```diff
@@ -2,51 +2,54 @@
...
 
-- Need super user rights (or root rights) to run.
+## Limitation
 
+- It Requires super user rights (or root rights) to run.
 
 ## Install keadm
 
-There're three ways to download a `keadm` binary
+There're three ways to download the `keadm` binary:
```
Comment (wbc6080) [inline docs/setup/install-with-keadm.md:16]: The component name here is `Keadm`, but `keadm` is used when executing such as `keadm join` on the command line and installing packages

File: docs/setup/install-with-keadm.md (hunk 3 above)
Comment (fujitatomoya) [reply docs/setup/install-with-keadm.md:40]: oh that was me misunderstanding, thanks for clarification!

**Review by fujitatomoya — APPROVED** [review]
lgtm

**Review by wbc6080 — COMMENTED** [review]
/lgtm
Thanks for your contribution
cc @Shelley-BaoYue @fisherxu

File: docs/setup/install-with-keadm.md [hunk 6]
```diff
@@ -2,51 +2,54 @@
...
+    wget https://github.com/kubeedge/kubeedge/releases/download/v1.17.0/keadm-v1.17.0-linux-amd64.tar.gz
+    tar -zxvf keadm-v1.17.0-linux-amd64.tar.gz
+    cp keadm-1.17.0-linux-amd64/keadm/keadm /usr/local/bin/keadm
     ```
-- Download from dockerhub KubeEdge official release image.
+
+2. Download from the official KubeEdge release image on Docker Hub.
 
   ```shell
   docker run --rm kubeedge/installation-package:v1.12.1 cat /usr/local/bin/keadm > /usr/local/bin/keadm && chmod +x /usr/local/bin/keadm
```
Comment (Shelley-BaoYue) [inline docs/setup/install-with-keadm.md:31]:
```suggestion
  docker run --rm kubeedge/installation-package:v1.17.0 cat /usr/local/bin/keadm > /usr/local/bin/keadm && chmod +x /usr/local/bin/keadm
```

File: docs/setup/install-with-keadm.md [hunk 7]
```diff
@@ -82,41 +87,46 @@ NAME                                   DESIRED   CURRENT   READY   AGE
 replicaset.apps/cloudcore-56b8454784   1         1         1       46s
 ```
 
-**IMPORTANT NOTE:**  
+**IMPORTANT NOTES:**  
+
+1. Set flags `--set key=value` for cloudcore helm chart could refer to [KubeEdge CloudCore Helm Charts README.md](https://github.com/kubeedge/kubeedge/blob/master/manifests/charts/cloudcore/README.md).
```
Comment (Shelley-BaoYue) [inline docs/setup/install-with-keadm.md:92]:
```suggestion
1. Set flags `--set key=value` for CloudCore helm chart could refer to [KubeEdge CloudCore Helm Charts README.md](https://github.com/kubeedge/kubeedge/blob/master/manifests/charts/cloudcore/README.md).
```

File: docs/setup/install-with-keadm.md [hunk 8]
```diff
@@ -266,30 +289,32 @@ Before deploying metrics-server , `kubectl logs` feature must be activated:
     ...
     ```
     
-    Then set all the iptables for multi cloudcore instances to every node that apiserver runs. The cloudcore ips and tunnel ports should be get from configmap above.
+    Then set all the iptables for multi cloudcore instances to every node that apiserver runs. The cloudcore ips and tunnel ports should be obtained from the configmap above.
```
Comment (Shelley-BaoYue) [inline docs/setup/install-with-keadm.md:292]:
```suggestion
    Then set all the iptables for multi CloudCore instances to every node that apiserver runs. The cloudcore ips and tunnel ports should be obtained from the configmap above.
```

File: docs/setup/install-with-keadm.md [hunk 9]
```diff
@@ -2,51 +2,54 @@
 title: Installing KubeEdge with Keadm
 sidebar_position: 3
 ---
-Keadm is used to install the cloud and edge components of KubeEdge. It is not responsible for installing K8s and runtime.
 
-Please refer [kubernetes-compatibility](https://github.com/kubeedge/kubeedge#kubernetes-compatibility) to get **Kubernetes compatibility** and determine what version of Kubernetes would be installed.
+Keadm is used to install the cloud and edge components of KubeEdge. It does not handle the installation of Kubernetes and its runtime environment.
```
Comment (Shelley-BaoYue) [inline docs/setup/install-with-keadm.md:6]:
```suggestion
Keadm is used to install the cloud and edge components of KubeEdge. It does not handle the installation of Kubernetes and its [runtime environment](https://kubeedge.io/docs/setup/prerequisites/runtime).
```

How about add a link with runtime? or we can put it at ##prerequisite?

Comment (1Shubham7) [reply docs/setup/install-with-keadm.md]: I have added the link with runtime. I don't think there's any point putting it at prerequisite.

**Review by Shelley-BaoYue — COMMENTED** [review]
/lgtm

**Comment by 1Shubham7** [comment]
@wbc6080 @fujitatomoya there have been some changes after you approved. Please approve this again.

**Review by fujitatomoya — APPROVED** [review]
thanks for iterating and being responsive. looks good to me.

**Comment by fujitatomoya** [comment]
i will leave this to @Shelley-BaoYue @wbc6080

**Review by wbc6080 — COMMENTED** [review]
/lgtm

**Comment by 1Shubham7** [comment]
@wbc6080 , @Shelley-BaoYue please approve this so that the PR gets merged. Also checkout the other improvement PRs that I have made :)

**Review by Shelley-BaoYue — COMMENTED** [review]
/approve

**Pull request merged**